### PR TITLE
ci: update PR process

### PR DESCRIPTION
**Pull request by htztomic**
Description of changes:
- add verify changefiles to PR checks for develop
- add license checker workflow
- fix: require PRs to be merged true to trigger workflow for main
Checklist:

* [ ] Have you successfully deployed to an AWS account with your changes?
* [ ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by insignias — DISMISSED** [review]

**Comment by insignias** [comment]
@htztomic Make sure to put this workflow in REQUIRED workflow list

File: .github/workflows/license-checker.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: license checker
+on: ['pull_request']
```
Comment (rsmayda) [inline .github/workflows/license-checker.yml:2]: only need to make a change if `package.json` is changed

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore

Comment (rsmayda) [reply .github/workflows/license-checker.yml:2]: I think you want it to be **package.json

**Comment by insignias** [comment]
@rsmayda @htztomic for license validation I already have a check: which can be updated for required license values and it already checks for it when rush build is triggered for example [here](https://github.com/awslabs/solution-spark-on-aws/runs/7783880872?check_suite_focus=true#step:3:1774)

File: .github/workflows/license-checker.yml (hunk 1 above)
Comment (insignias) [reply .github/workflows/license-checker.yml:2]: @rsmayda @htztomic for license validation I already have a check: which can be updated for allowed license values and it already checks for it when rush build is triggered for example [here](https://github.com/awslabs/solution-spark-on-aws/runs/7783880872?check_suite_focus=true#step:3:1774)

Comment (htztomic) [reply .github/workflows/license-checker.yml:2]: @insignias this is checking if a project's dependencies have the correct license value

Comment (insignias) [reply .github/workflows/license-checker.yml:2]: ahh ok !

**Review by insignias — DISMISSED** [review]

File: .github/workflows/create-release-branch.yml [hunk 2]
```diff
@@ -15,4 +15,9 @@ jobs:
           uuid=`uuidgen`
           echo 'Creating release branch'
           git checkout -b release/$uuid
+          
+          echo 'Rush version to bump updates'
+          node common/scripts/install-run-rush.js version --bump -b release/$uuid --ignore-git-hooks
+
+          echo 'Push changes'
```
Comment (insignias) [inline .github/workflows/create-release-branch.yml:22]: Do we need push here ?

**Review by insignias — APPROVED** [review]

**Review by rsmayda — APPROVED** [review]

**Pull request merged**